software/controller/lib/core/commfail_alarm.cpp

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <stdint.h>
+#include "debug.h"
+#include "hal.h"
+#include "alarm.h"
+#include "commfail_alarm.h"
+
+extern alarm_cb_t alarms[];
```

Comment:
Avoid static initialization with global variables. It does not have predictable behavior. This should explicitly initialize this in main somehow, or use a singleton pattern.